expression/builtin_math.go

```diff
@@ -702,7 +702,14 @@ func (b *builtinFloorIntToDecSig) Clone() builtinFunc {
 // See https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_floor
 func (b *builtinFloorIntToDecSig) evalDecimal(row types.Row) (*types.MyDecimal, bool, error) {
 	val, isNull, err := b.args[0].EvalInt(b.ctx, row)
-	return types.NewDecFromUint(uint64(val)), isNull, errors.Trace(err)
+	if isNull || err != nil {
+		return nil, true, errors.Trace(err)
+	}
+
+	if mysql.HasUnsignedFlag(b.args[0].GetType().Flag) || val >= 0 {
```

Comment:
Why we need to check `val > 0` when it has `unsigned` flag

Reply:
Oh.. `val >= 0` is checked only when `mysql.HasUnsignedFlag(b.args[0].GetType().Flag)` is false